### PR TITLE
[#8668] [Platform] [API] Tables controller error handling. 

### PR DESCRIPTION
**Pull request by jitendra-12113**
**Description:**
Use the new global exception handling in the TablesController.

**Test Plan:**

Its mostly refactor. Unit tests are updated.
Run server and test basic operations.

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java [hunk 1]
```diff
@@ -97,17 +98,8 @@ public Result create(UUID customerUUID, UUID universeUUID) {
       resultNode.put("taskUUID", taskUUID.toString());
       auditService().createAuditEntry(ctx(), request(), Json.toJson(formData.data()));
       return Results.status(OK, resultNode);
-    } catch (NullPointerException e) {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:100]: You need to see where NPE is coming from.

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java [hunk 2]
```diff
@@ -97,17 +98,8 @@ public Result create(UUID customerUUID, UUID universeUUID) {
...
-      LOG.error("Error creating table", e);
-      // This error isn't useful at all, why send a NullPointerException as api response?
-      return ApiResponse.error(BAD_REQUEST, "NullPointerException");
-    } catch (RuntimeException e) {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:104]: Where is RuntimeEx coming from?

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java [hunk 3]
```diff
@@ -97,17 +98,8 @@ public Result create(UUID customerUUID, UUID universeUUID) {
...
-      // TODO: Error prone - remove this
-      LOG.error("Error creating table", e);
-      return ApiResponse.error(BAD_REQUEST, e.getMessage());
     } catch (Exception e) {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:101]: Exceptions should result in 500. Once you figure out the root cause of the other two; you should make them cause YWSE. Then you wont need try catch block.

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java [hunk 4]
```diff
@@ -250,10 +235,7 @@ public Result universeList(UUID customerUUID, UUID universeUUID) {
       }
       return ok(resultNode);
     } catch (Exception e) {
-      LOG.error("Failed to get list of tables in universe " + universeUUID, e);
-      return ApiResponse.error(INTERNAL_SERVER_ERROR, e.getMessage());
-    } finally {
-      ybService.closeClient(client, masterAddresses);
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:256]: You do need finally block that closes the client.

Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:256]: So just do try {} finally {}  ... no need to catch

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java [hunk 5]
```diff
@@ -250,10 +235,7 @@ public Result universeList(UUID customerUUID, UUID universeUUID) {
...
+      throw new YWServiceException(BAD_REQUEST, e.getMessage());
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:238]: why BAD_REQUEST? this used to be 500 before.

File: managed/src/test/java/com/yugabyte/yw/controllers/TablesControllerTest.java [hunk 6]
```diff
@@ -214,9 +214,14 @@ public void testCreateCassandraTableWithInvalidParams() {
     String url =
         "/api/customers/" + customer.uuid + "/universes/" + universe.universeUUID + "/tables";
     ObjectNode emptyJson = Json.newObject();
```
Comment (sb-yb) [inline managed/src/test/java/com/yugabyte/yw/controllers/TablesControllerTest.java:216]: Ha!. so this is what that NPE is about.

File: managed/src/test/java/com/yugabyte/yw/controllers/TablesControllerTest.java [hunk 7]
```diff
@@ -214,9 +214,14 @@ public void testCreateCassandraTableWithInvalidParams() {
...
-    String errorString = "NullPointerException";
+    String errorString = "null";
 
-    Result result = FakeApiHelper.doRequestWithAuthTokenAndBody(method, url, authToken, emptyJson);
```
Comment (sb-yb) [inline managed/src/test/java/com/yugabyte/yw/controllers/TablesControllerTest.java:219]: And to make this test happy you changed all exceptions to cause BAD_REQUEST

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java (hunk 1 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:100]: taskParams.tableDetails is coming as null :) -> line no 71
Will add null check for this one and throw Bad_request.

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java (hunk 2 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:104]: I think this is not required.  Just removed try-catch block

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java (hunk 3 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:101]: Done.

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java (hunk 4 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:256]: Done.

File: managed/src/test/java/com/yugabyte/yw/controllers/TablesControllerTest.java (hunk 6 above)
Comment (jitendra-12113) [reply managed/src/test/java/com/yugabyte/yw/controllers/TablesControllerTest.java:216]: Yeah :)

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java (hunk 4 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:256]: We would need to handle the Exception
client.getTablesList() - this is throwing exception explicitly in the method -> line no 228

Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:256]: I wish that client threw more meaningful exceptions. But thats outside our cleanup. Just use try-catch-finally for that part of the code then. Extract method and call that:
```
private ListTablesResponse listTables() {
YBClient client = null;    
try {
      client = ybService.getClient(masterAddresses, certificate);
      ListTablesResponse response = client.getTablesList();
} catch (Exception e) {
      throw new YWServiceException(INTERNAL_SERVER_ERROR, e.getMessage());
} finally {
ybService.closeClient(client, masterAddresses);
}
}
```

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java (hunk 4 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:256]: Done. Thanks!

File: managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java [hunk 8]
```diff
@@ -206,33 +206,38 @@ public Result universeList(UUID customerUUID, UUID universeUUID) {
     }
 
     String certificate = universe.getCertificateNodetoNode();
+    ListTablesResponse response = listTables(masterAddresses, certificate);
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/TablesController.java:209]: nit: listTablesOrBadRequest

**Review by sb-yb — APPROVED** [review]

**Pull request merged**